zcash_client_backend/src/data_api/chain.rs

```diff
@@ -403,16 +396,25 @@ where
     ///
     /// This method should return an error if contiguous blocks cannot be read from the cache,
     /// indicating there are blocks missing.
-    async fn read(&self, range: &ScanRange) -> Result<Vec<CompactBlock>, Self::Error>;
+    async fn read<WalletErrT>(
+        &self,
+        range: &ScanRange,
+    ) -> Result<Vec<CompactBlock>, error::Error<WalletErrT, Self::Error>>;
```

Comment:
I agree, my original PR for this work actually removed the wallet errors from BlockSource instead of adding them to BlockCache. However, I am not sure of the intention behind WalletErrT so I wanted to leave this undisrupted.

Reply:
I don't see why this should be necessary either. If you can refactor it to just use `Self::Error`, please do so.

Reply:
The `WalletErrT` on `BlockSource::with_blocks` is currently used to get wallet errors out of the `with_block` closure. I think the current approach of hard-coding `error::Error` into the interface can however be removed, and I will do so because `BlockCache` definitely shouldn't need `WalletErrT`.

Reply:
Done!